packages/blade/src/components/Radio/RadioIcon/RadioIconWrapperStyles.ts

```diff
@@ -0,0 +1,76 @@
+import type { CSSObject } from 'styled-components';
+import type { RadioIconProps } from './RadioIcon';
+import type { Theme } from '~components/BladeProvider';
+import { getIn, getPlatformType, makeMotionTime, makeSize, makeSpace } from '~utils';
+
+const variants = {
+  default: {
+    border: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.400',
+    },
+    background: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.200',
+    },
+  },
+  disabled: {
+    border: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+    background: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+  },
+  negative: {
+    border: {
+      checked: 'colors.feedback.border.negative.highContrast',
+      unchecked: 'colors.feedback.border.negative.highContrast',
+    },
+    background: {
+      checked: 'colors.feedback.background.negative.lowContrast',
+      unchecked: 'colors.feedback.background.negative.lowContrast',
+    },
+  },
+};
+
+export type RadioRectProps = RadioIconProps;
+
+const getRadioIconWrapperStyles = ({
+  theme,
+  isChecked,
+  isDisabled,
+  isNegative,
+}: RadioRectProps & { theme: Theme }): CSSObject => {
+  const isReactNative = getPlatformType() === 'react-native';
+  let variant: 'default' | 'disabled' | 'negative' = 'default';
+  if (isDisabled) variant = 'disabled';
+  if (isNegative) variant = 'negative';
+  const checked = isChecked ? 'checked' : 'unchecked';
+  const backgroundColor = variants[variant].background[checked];
+  const borderColor = variants[variant].border[checked];
+
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '16px',
+    height: '16px',
+    borderWidth: '1px',
+    borderStyle: 'solid',
+    margin: '0px',
+    marginTop: '3px',
```

Comment:
Margin of 3px? Where is this?

Reply:
Not a token from figma. It's to vertical align the Icon to text. We can't use flex since It has to stay in the center of text not the whole wrapper.

Reply:
This could be troublesome. 3px might not stay true to align it to the center in various circumstances - platform (android, ios, web), pixel density, a11y font changes, etc. Have you verified the alignment on Android as well as iOS?

Reply:
I see yeah could be,  I don't see any difference in native. Anyway are we doing pixel density calculation in any of our spacing tokens?

Reply:
No but we shouldn't use margin to center align it. We should rely on some form of layout